formats/json-tests/commonTest/src/kotlinx/serialization/json/JsonExponentTest.kt

```diff
@@ -0,0 +1,79 @@
+package kotlinx.serialization.json
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.test.*
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class JsonExponentTest : JsonTestBase() {
+    @Serializable
+    data class SomeData(val count: Long)
+    @Serializable
+    data class SomeDataDouble(val count: Double)
+
+    @Test
+    fun testExponentDecodingPositive() = parametrizedTest {
+        val decoded =  Json.decodeFromString<SomeData>("""{ "count": 23e11 }""", it)
+        assertEquals(2300000000000, decoded.count)
+    }
+
+    @Test
+    fun testExponentDecodingNegative() = parametrizedTest {
+        val decoded = Json.decodeFromString<SomeData>("""{ "count": -10E1 }""", it)
+        assertEquals(-100, decoded.count)
+    }
+
+    @Test
+    fun testExponentDecodingTruncatedDecimal() = parametrizedTest {
+        val decoded = Json.decodeFromString<SomeData>("""{ "count": -1E-1 }""", it)
+        assertEquals(0, decoded.count)
+    }
+
+    @Test
+    fun testExponentDecodingPositiveDouble() = parametrizedTest {
+        val decoded = Json.decodeFromString<SomeDataDouble>("""{ "count": 1.5E1 }""", it)
```

Comment:
Should cases like this one be supported when decoding into a long | int too? and if so, should it be truncated, rounded...?

Reply:
I think this can be left as is in order not to complicate the code. We can always add it later, if there's a request to do so. Original 'parsing exponent to Long' request was about minifying, and I hope minifier just don't insert random dots in the number.